compiler/rustc_session/src/config.rs

```diff
@@ -580,7 +580,7 @@ pub enum ResolveDocLinks {
 /// *Do not* switch `BTreeMap` out for an unsorted container type! That would break
 /// dependency tracking for command-line arguments. Also only hash keys, since tracking
 /// should only depend on the output types, not the paths they're written to.
-#[derive(Clone, Debug, Hash, HashStable_Generic)]
+#[derive(Clone, Debug, Hash, HashStable_Generic, Encodable, Decodable)]
```

Comment:
Could this cause us to leak host info into build artifacts?

Reply:
The rlink file already contains absolute file paths. `rustc -Zlink-only foo.rlink` is supposed to be called immediately after all dependencies of a `rustc -Zlink-only` invocation are finished. This has to be done on the same machine and can only be done once because all temporary files that are shared between the two rustc invocations have unpredictable names and are removed by `rustc -Zlink-only`.